pennylane/wires.py

```diff
@@ -50,6 +59,15 @@ def _process(wires):
         # of considering the elements of iterables as wire labels.
         wires = [wires]
 
+    if jax_available:
+        if isinstance(wires, jax.numpy.ndarray) and not isinstance(wires, jax.core.Tracer):
+            wires = tuple(wires.tolist() if wires.ndim > 0 else (wires.item(),))
```

Comment:
The `if wires.ndim > 0 else (wires.item(),))` is inserted so that something like `jax.numpy.array(0)` works. Notice that this currently raises an error for numpy arrays as it is not hashable, but I don't see why we should not make it work here for JAX arrays.

Reply:
What do you mean it's not hashable? Shouldn't the elements of the numpy array be hashable?

Reply:
We don't need them to be hashable though, right? That's the point of treating them as an iterable

Reply:
> jax.numpy.array([0]) is hashable but jax.numpy.array(0) isn't (same goes for standard numpy arrays)

Also I could not confirm this statement to be true 🤔 

```
>>> hash(jnp.array([5]))
TypeError: unhashable type: 'jaxlib.xla_extension.ArrayImpl'
>>> hash(jnp.array(5))
TypeError: unhashable type: 'jaxlib.xla_extension.ArrayImpl'

Reply:
Sorry, bad choice of words. The correct statement should be:

_`qml.wires.Wires(np.array(0))` generates a `WireError: Wires must be hashable` error, but `qml.wires.Wires(np.array([0]))` does not._

The addition `if wires.ndim > 0 else (wires.item(),)` in this PR ensures that both `qml.wires.Wires(jax.numpy.array(0))` and `qml.wires.Wires(jax.numpy.array([0]))` are accepted. If we explicitly don't want the former case to work we can simply remove the addition

Reply:
No I totally agree with that bit of code, was just trying to understand your comment about numpy arrays.

Reply:
I updated the comment (hopefully it is less ambiguous now). I'll add this into the PR description before merging as a future reference